bundles/com.espressif.idf.serial.monitor/src/com/espressif/idf/serial/monitor/server/SocketServerHandler.java

```diff
@@ -0,0 +1,79 @@
+/*******************************************************************************
+ * Copyright 2021 Espressif Systems (Shanghai) PTE LTD. All rights reserved.
+ * Use is subject to license terms.
+ *******************************************************************************/
+package com.espressif.idf.serial.monitor.server;
+
+import java.io.IOException;
+import java.net.ServerSocket;
+import java.util.Queue;
+
+import org.eclipse.core.resources.IProject;
+
+import com.espressif.idf.core.util.SDKConfigJsonReader;
+
+/**
+ * Socket server handler responsible for starting and handling the socket server
+ * 
+ * @author Ali Azam Rana
+ *
+ */
+public class SocketServerHandler
+{
+	private static TerminalWebSocketServer terminalWebSocketServer;
+
+	public int startServer() throws Exception
+	{
+		int port = findPort();
+		terminalWebSocketServer = new TerminalWebSocketServer(port);
+		terminalWebSocketServer.start();
+		return terminalWebSocketServer.getPort();
+	}
+
+	public void broadcastMessageToClients(String message)
+	{
+		terminalWebSocketServer.broadcast(message);
+	}
+
+	public Queue<String> getMessagesQueue()
+	{
+		return terminalWebSocketServer.getMessagesReceivedQueue();
+	}
+
+	private int findPort() throws IOException
+	{
+		ServerSocket serverSocket = new ServerSocket(0);
+		int port = serverSocket.getLocalPort();
+		serverSocket.close();
+		return port;
+	}
+
+	public void stopServer() throws Exception
+	{
+		terminalWebSocketServer.stop(500);
+	}
+
+	public static int getServerPort()
+	{
+		return terminalWebSocketServer.getPort();
+	}
+
+	public static boolean needSocketServer(IProject project)
+	{
+		String gdbStubPanicInfo = getGdbPanicStubInfo(project);
+		if (gdbStubPanicInfo.equalsIgnoreCase("null") || gdbStubPanicInfo.equalsIgnoreCase("false")) //$NON-NLS-1$ $NON-NLS-2$
+		{
+			return false;
+		}
+		else
+		{
+			return true;
+		}
+	}
```

Comment:
Could we replace the above code with `return Boolean.valueOf(gdbStubPanicInfo).booleanValue())` - this will take care of null check and return the appropriate boolean value

Reply:
Done